coderd/notifications/metrics_test.go

```diff
@@ -0,0 +1,342 @@
+package notifications_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	promtest "github.com/prometheus/client_golang/prometheus/testutil"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/serpent"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbmem"
+	"github.com/coder/coder/v2/coderd/database/dbtestutil"
+	"github.com/coder/coder/v2/coderd/notifications"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestMetrics(t *testing.T) {
+	t.Parallel()
+
+	// setup
+	if !dbtestutil.WillUsePostgres() {
```

Comment:
Is it because of pubsub or something else?

Reply:
A lot of the business logic is implemented in the database layer.
I'm requiring postgres here because the full functionality is required (retries, etc).
I'll clarify this inline.